packages/nextjs/src/middleware/locale-middleware.ts

```diff
@@ -0,0 +1,88 @@
+/* eslint-disable @typescript-eslint/no-non-null-asserted-optional-chain */
+
+import { NextResponse, NextRequest } from 'next/server';
+import { debug } from '@sitecore-content-sdk/core';
+import { getLocaleRewrite } from '@sitecore-content-sdk/core/i18n';
+import { MiddlewareBase, MiddlewareBaseConfig, LOCALE_HEADER_NAME } from './middleware';
+
+export type LocaleMiddlewareConfig = MiddlewareBaseConfig & {
+  locales: string[];
+};
+
+export class LocaleMiddleware extends MiddlewareBase {
```

Comment:
Let's add jsdoc, it should be clear that this middleware is suppose to be used in App Router only